File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);
        int n=strs.length;

        String common="";

        String f=strs[0];
        String l=strs[n-1];
        int m=Math.min(f.length(),l.length());
        int i=0;
        while(i<m && f.charAt(i)==l.charAt(i)){
            common+=f.charAt(i);
            i++;
        }
        return common;
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        int n=strs.size();
        if(n==1) return strs[0];
        sort(strs.begin(),strs.end());
        string ans="";
        string f=strs[0];
        string l=strs[n-1];
        int i=0;
        int k=min(f.size(),l.size());
        while(i<k && f[i]==l[i] ){
            ans+=f[i];
            i++;
        }
        return ans;
    }
};